CMake/External_Caffe_Segnet.cmake

```diff
@@ -0,0 +1,356 @@
+if (WIN32)
+  message( FATAL_ERROR "Caffe-Segnet is not yet supported on windows" )
+  return()
+endif()
+
+
+set(allOk True)
+set(errorMessage)
+
+option(AUTO_ENABLE_CAFFE_SEGNET_DEPENDENCY "Automatically turn on all caffe segnet dependencies if caffe segnet is enabled" OFF)
+if(fletch_ENABLE_Caffe_Segnet AND AUTO_ENABLE_CAFFE_SEGNET_DEPENDENCY)
+  #Snappy is needed by LevelDB and ZLib is needed by HDF5
+  if(WIN32)
+    set(dependency Boost ZLib OpenCV HDF5)
+  else()
+    set(dependency Boost GFlags GLog ZLib HDF5 Snappy LevelDB LMDB OpenCV Protobuf)
+  endif()
+
+  if(NOT APPLE AND NOT WIN32)
+    list(APPEND dependency OpenBLAS)
+  endif()
+
+  set(OneWasOff FALSE)
+
+  foreach (_var IN LISTS dependency)
+    get_property(currentHelpString CACHE "fletch_ENABLE_${_var}" PROPERTY HELPSTRING)
+    set(fletch_ENABLE_${_var} ON CACHE BOOL ${currentHelpString} FORCE)
+    if(NOT TARGET ${_var})
+      include(External_${_var})
+    endif()
+  endforeach()
+endif()
+
+function(addCaffeSegnetDendency depend version)
+  if(NOT fletch_ENABLE_${depend} )
+    find_package(${depend} ${version} QUIET)
+    string(TOUPPER "${depend}" dependency_name_upper)
+    if(NOT ${depend}_FOUND AND NOT ${dependency_name_upper}_FOUND)
+      message("${depend} is needed")
+      set(allOk False PARENT_SCOPE)
+      return()
+    endif()
+    message("Warning: Using system library for ${depend}")
+  else() #need to make sure library is built before caffe segnet
+    set(Caffe_Segnet_DEPENDS ${Caffe_Segnet_DEPENDS} ${depend} PARENT_SCOPE)
+  endif()
+  add_package_dependency(
+    PACKAGE Caffe_Segnet
+    PACKAGE_DEPENDENCY ${depend}
+    PACKAGE_DEPENDENCY_ALIAS ${depend}
+    )
+endfunction()
+
+# Check for dependencies.
+if(NOT WIN32) # Win32 build takes care of most dependencies automatically
+  addCaffeSegnetDendency(GFlags "")
+  addCaffeSegnetDendency(GLog "")
```

Comment:
Both GLog and GTest are available on Window now and included in Caffe Windows.

Reply:
Ok, but FYI
We still need to go through the actual code base and make it windows compatible much like the kwiver gitlab caffe version we have.